src/hooks/codeImport.hooks.ts

```diff
@@ -0,0 +1,60 @@
+import { ContractLanguage } from '@/interfaces/workspace.interface';
+import { decodeBase64 } from '@/utility/utils';
+import { useRouter } from 'next/router';
+import useFileTab from './fileTabs.hooks';
+import { baseProjectPath, useProject } from './projectV2.hooks';
+
+const useCodeImport = () => {
+  const router = useRouter();
+  const { createProject } = useProject();
+  const { open: openTab } = useFileTab();
+
+  const importEncodedCode = async (
+    code: string,
+    language: ContractLanguage,
+  ) => {
+    const fileExtension = language === 'func' ? 'fc' : language;
+    const defaultFileName = `main.${fileExtension}`;
+
+    await removeImportParams();
+
+    await createProject({
+      name: 'temp',
+      language,
+      template: 'import',
+      file: null,
+      defaultFiles: [
+        {
+          id: '',
+          parent: null,
+          path: defaultFileName,
+          type: 'file' as const,
+          name: defaultFileName,
+          content: decodeBase64(code),
+        },
+      ],
+      isTemporary: true,
+    });
+
+    openTab(defaultFileName, `${baseProjectPath}/temp/${defaultFileName}`);
+  };
+
+  const removeImportParams = async () => {
+    // Remove all query params related to importing code
+    const keysToRemove = ['code', 'lang', 'importURL', 'name'];
+    const finalQueryParam = Object.fromEntries(
+      Object.entries(router.query).filter(
+        ([key]) => !keysToRemove.includes(key),
+      ),
+    );
+
+    await router.replace({ query: finalQueryParam });
+  };
+
+  return {
+    importEncodedCode,
```

Comment:
If these event handlers end up passed not to a `<div>`, but to a component, this will trigger rerender of the component with no regard to memoization, because reference to these functions always changes.

Please, use `useCallback`.